### DISCUSSION TITLE
How can I import osgVolume from OSG?

### DISCUSSION THREAD

**Discussion by jepessen** [microsoft/vcpkg#23326] [General]
I want to use the osgVolume library from OpenSceneGraph, in Windows.
The library was installed correctly, but I cannot find the library. The full path of the library is

```
E:\Projects\project\build\Debug\vcpkg_installed\x64-windows\lib
```

And I'm trying to find it with:

```
find_library (OSGVOLUME_RELEASE_LIB osgVolume  PATHS ${_VCPKG_INSTALLED_DIR}/${VCPKG_TARGET_TRIPLET}/lib       NO_DEFAULT_PATH)
```

But the content of `OSGVOLUME_RELEASE_LIB` is (printed with a message):

```
E:/Projects/project/build/Debug/vcpkg_installed/x64-windows/include
```

If I try to find the debug library, it works:

```
find_library (OSGVOLUME_DEBUG_LIB   osgVolumed PATHS ${_VCPKG_INSTALLED_DIR}/${VCPKG_TARGET_TRIPLET}/debug/lib NO_DEFAULT_PATH)
OSGVOLUME_DEBUG_LIB    is E:/Projects/project/build/Debug/vcpkg_installed/x64-windows/debug/lib/osgVolumed.lib
```

The `_VCPKG_INSTALLED_DIR` and `VCPKG_TARGET_TRIPLET` variables are correct, and `${_VCPKG_INSTALLED_DIR}/${VCPKG_TARGET_TRIPLET}` is

```
E:/Projects/project/build/Debug/vcpkg_installed/x64-windows
```

In `include` dir there's a header folder named `osgVolume`, maybe it can cause the confusion.

How can I find the correct file?

**Comment by Osyotr** [comment]
Why don't you use built-in module?
https://cmake.org/cmake/help/v3.23/module/FindOpenSceneGraph.html

**Comment by dg0yt** [comment]
> And I'm trying to find it with:
> 
> ```
> find_library (OSGVOLUME_RELEASE_LIB osgVolume  PATHS ${_VCPKG_INSTALLED_DIR}/${VCPKG_TARGET_TRIPLET}/lib       NO_DEFAULT_PATH)
> ```
> 
> But the content of `OSGVOLUME_RELEASE_LIB` is (printed with a message):
> 
> ```
> E:/Projects/project/build/Debug/vcpkg_installed/x64-windows/include
> ```

I would take this as indication that the *cache variable* `OSGVOLUME_RELEASE_LIB` is already set before calling `find_library`. Check if you use in `OSGVOLUME_RELEASE_LIB` in parameters to cmake or in another, earlier call.
